state/tx_options.go

```diff
@@ -0,0 +1,168 @@
+package state
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/celestiaorg/celestia-app/pkg/user"
+	apptypes "github.com/celestiaorg/celestia-app/x/blob/types"
+)
+
+const (
+	DefaultPrice float64 = -1.0
+	// gasMultiplier is used to increase gas limit in case if tx has additional options.
+	gasMultiplier = 1.1
+)
+
+func NewTxOptions(attributes ...Attribute) *TxOptions {
+	options := &TxOptions{gasPrice: DefaultPrice}
+	for _, attr := range attributes {
+		attr(options)
+	}
+	return options
+}
+
+// TxOptions specifies additional options that will be applied to the Tx.
+// Implements `Option` interface.
+type TxOptions struct {
+	// gasPrice represents the amount to be paid per gas unit.
+	// Negative gasPrice means user want us to use the minGasPrice
+	// defined in the node.
+	gasPrice float64
+	// since gasPrice can be 0, it is necessary to understand that user explicitly set it.
+	isGasPriceSet bool
```

Comment:
I'm guessing it's either negative`gasPrice` or `isGasPriceSet==false` which indicates automatic estimating of gas price

Reply:
`isGasPriceSet` helps to detect whether the user explicitly sets 0 or instantiates an empty `TxOptions{}` struct. I agree that comments look confusing. Will change them accordingly